<a href="https://colab.research.google.com/github/KoushikDas161/iris_pandas_matplotlib/blob/main/iris_pandas_matplotlib.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Setup: import pandas

In [ ]:
import pandas as pd

Load the Iris dataset into a DataFrame

In [ ]:
df = pd.read_csv('/content/Iris.csv')

In [ ]:
df

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
0,1,5.1,3.5,1.4,0.2,Iris-setosa
1,2,4.9,3.0,1.4,0.2,Iris-setosa
2,3,4.7,3.2,1.3,0.2,Iris-setosa
3,4,4.6,3.1,1.5,0.2,Iris-setosa
4,5,5.0,3.6,1.4,0.2,Iris-setosa
...,...,...,...,...,...,...
145,146,6.7,3.0,5.2,2.3,Iris-virginica
146,147,6.3,2.5,5.0,1.9,Iris-virginica
147,148,6.5,3.0,5.2,2.0,Iris-virginica
148,149,6.2,3.4,5.4,2.3,Iris-virginica


Check basic info
Use `.shape`, `.columns`, `.info()`, and `.describe()` to understand the dataset.

Row selection: iloc (position-based)

In [ ]:
df.iloc[:5, 2]

,SepalWidthCm
0,3.5
1,3.0
2,3.2
3,3.1
4,3.6


Row selection: loc (label-based) (sepal length(cm), species)

In [ ]:
df.loc[:5, 'SepalWidthCm']

,SepalWidthCm
0,3.5
1,3.0
2,3.2
3,3.1
4,3.6
5,3.9


Boolean filtering

In [ ]:
#Show all setosa rows
df[df["Species"]=="Iris-setosa"].head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
0,1,5.1,3.5,1.4,0.2,Iris-setosa
1,2,4.9,3.0,1.4,0.2,Iris-setosa
2,3,4.7,3.2,1.3,0.2,Iris-setosa
3,4,4.6,3.1,1.5,0.2,Iris-setosa
4,5,5.0,3.6,1.4,0.2,Iris-setosa


**Task:** Filter rows where:
- species is `Iris-virginica`
- petal length (cm) is greater than 5.5

Store it in `df_v` and show first 5 rows.

 Assign new columns
* Add petal area(petal length * petal width), sepal area columns(sepal length * sepal width)
* show first 5 rows



In [ ]:
df['petal_area']= df['PetalLengthCm']* df['PetalWidthCm']
df.head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species,petal_area
0,1,5.1,3.5,1.4,0.2,Iris-setosa,0.28
1,2,4.9,3.0,1.4,0.2,Iris-setosa,0.28
2,3,4.7,3.2,1.3,0.2,Iris-setosa,0.26
3,4,4.6,3.1,1.5,0.2,Iris-setosa,0.30
4,5,5.0,3.6,1.4,0.2,Iris-setosa,0.28


**Task:** Add a boolean column called `big_petal` where petal length (cm) > 4.5.
Then display `petal length (cm)`, `species`, `big_petal`.

Summary Functions and `map` / `apply`

Summary statistics (sepal length mean, petal width median, value_counts())

In [ ]:
df['sepal_length_rounded']= df['SepalLengthCm'].map(lambda x: round(x, 1))

In [ ]:
df.head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species,petal_area,sepal_length_rounded,setosa_code
0,1,5.1,3.5,1.4,0.2,Iris-setosa,0.28,5.1,1
1,2,4.9,3.0,1.4,0.2,Iris-setosa,0.28,4.9,1
2,3,4.7,3.2,1.3,0.2,Iris-setosa,0.26,4.7,1
3,4,4.6,3.1,1.5,0.2,Iris-setosa,0.30,4.6,1
4,5,5.0,3.6,1.4,0.2,Iris-setosa,0.28,5.0,1


In [ ]:
df['setosa_code']= df['Species'].map({
    'Iris-setosa': 1,
    'Iris-virginica': 2,
    'Iris-versicolor':3
})
df.head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species,petal_area,sepal_length_rounded,setosa_code
0,1,5.1,3.5,1.4,0.2,Iris-setosa,0.28,5.1,1
1,2,4.9,3.0,1.4,0.2,Iris-setosa,0.28,4.9,1
2,3,4.7,3.2,1.3,0.2,Iris-setosa,0.26,4.7,1
3,4,4.6,3.1,1.5,0.2,Iris-setosa,0.30,4.6,1
4,5,5.0,3.6,1.4,0.2,Iris-setosa,0.28,5.0,1


 `map()` for converting a single column to uppercase
 * hint: lambda x: x.upper()

 Grouping and Sorting

In [ ]:
df.head(2)

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species,petal_area,sepal_length_rounded,setosa_code
0,1,5.1,3.5,1.4,0.2,Iris-setosa,0.28,5.1,1
1,2,4.9,3.0,1.4,0.2,Iris-setosa,0.28,4.9,1


Calculate Mean feature values per species
* using df.groupby("Species")

In [ ]:
df.groupby("Species")[['PetalLengthCm', 'PetalWidthCm']].mean()

,PetalLengthCm,PetalWidthCm
Species,,
Iris-setosa,1.464,0.244
Iris-versicolor,4.260,1.326
Iris-virginica,5.552,2.026


Multiple aggregations with `.agg()`

Sorting
Sort species by mean petal area (descending)

In [ ]:
df["petal_area"].sort_values(ascending= False).head()

,petal_area
118,15.87
109,15.25
100,15.00
117,14.74
144,14.25


In [ ]:
df.sort_values(by="petal_area", ascending= False).head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species,petal_area,sepal_length_rounded,setosa_code
118,119,7.7,2.6,6.9,2.3,Iris-virginica,15.87,7.7,2
109,110,7.2,3.6,6.1,2.5,Iris-virginica,15.25,7.2,2
100,101,6.3,3.3,6.0,2.5,Iris-virginica,15.00,6.3,2
117,118,7.7,3.8,6.7,2.2,Iris-virginica,14.74,7.7,2
144,145,6.7,3.3,5.7,2.5,Iris-virginica,14.25,6.7,2


**Task:** Sort the original `df` by `species` (A→Z) and within each species, sort by `petal length (cm)` (largest first).
Show first 10 rows.

check for missing values
* isnull().sum()
* if missing: dropna()

Exercise

Split dataframes into two parts and recombine using pd.concat():

In [ ]:
df.shape

(150, 9)

In [ ]:
part1 = df[75:]
part2= df[:75]
df_new= pd.concat([part1, part2])

In [ ]:
df_new.head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species,petal_area,sepal_length_rounded,setosa_code
75,76,6.6,3.0,4.4,1.4,Iris-versicolor,6.16,6.6,3
76,77,6.8,2.8,4.8,1.4,Iris-versicolor,6.72,6.8,3
77,78,6.7,3.0,5.0,1.7,Iris-versicolor,8.50,6.7,3
78,79,6.0,2.9,4.5,1.5,Iris-versicolor,6.75,6.0,3
79,80,5.7,2.6,3.5,1.0,Iris-versicolor,3.50,5.7,3



Filter: species = setosa and petal length (cm) < 1.5. Show first 5 rows.


Compute mean and std of sepal length (cm) for each species using `groupby` + `agg`.